presto-main/src/main/java/com/facebook/presto/execution/scheduler/SectionExecutionFactory.java

```diff
@@ -342,8 +342,9 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
 
                 BucketNodeMap bucketNodeMap;
                 List<InternalNode> stageNodeList;
-                if (plan.getFragment().getRemoteSourceNodes().stream().allMatch(node -> node.getExchangeType() == REPLICATE)) {
-                    // no non-replicated remote source
+                if (plan.getFragment().getRemoteSourceNodes().stream().allMatch(node -> node.getExchangeType() == REPLICATE)
+                        && schedulingOrder.stream().allMatch(id -> plan.getFragment().getStageExecutionDescriptor().isScanGroupedExecution(id))) {
+                    // no non-replicated remote source and all scans are grouped
```

Comment:
A little question, do we need this change? Should a fragment with broadcast-only remote sources but couldn't support dynamic lifespan scheduling still go through this path?

Reply:
oh, I think you are right that we don't need this! The case you describe does here, but the stage execution descriptor won't be dynamicLifespanSchedule, and so we'll get a non-dynamic bucket node map in the two lines below).   Now I see that the else path is for stages that need a nodePartitionMap, which I believe is only needed for a partitioned remote source.

Reply:
fixed